Kirby level progress

AnonymousRandomPerson opened the pull request:
First iteration of Kirby level.
-Added functionality for negotiating gentle slopes.
-Added Waddle Doo clone enemy (sprite totally not stolen from NID).
-Initial platform layout.

AnonymousRandomPerson committed:
Start of Kirby level

-Grouped all necessary managers together.
-Primitive sprites for Kirby level.
-Some adaptations for slope physics.

AnonymousRandomPerson committed:
Slope physics

AnonymousRandomPerson committed:
Merge remote-tracking branch 'JonathanHunter/master' into kirbylevel

AnonymousRandomPerson committed:
Waddle Doo clone

AnonymousRandomPerson committed:
Merge remote-tracking branch 'JonathanHunter/master' into kirbylevel

AnonymousRandomPerson committed:
Subpar beam enemy sprite

AnonymousRandomPerson committed:
Merge remote-tracking branch 'JonathanHunter/master' into kirbylevel

AnonymousRandomPerson committed:
Merge remote-tracking branch 'JonathanHunter/master' into kirbylevel

AnonymousRandomPerson committed:
Filled out Kirby level environment

AnonymousRandomPerson committed:
Reverted test scene destructibilty

JonathanHunter commented on Assets/Scripts/Enemy/Enemies/BeamEnemy.cs:
```diff
@@ -0,0 +1,78 @@
+            else
+            {
+                Vector2 forward = GetForward();
+                RaycastHit2D hit = Physics2D.Raycast(front.position, forward, speed * Time.deltaTime, ~(1 << 8 | 1 << 10 | 1 << 12));
```
use TouchingSomething() and Turn() to do this

JonathanHunter commented:
Use the game manager and input objects from the title screen to avoid errors

AnonymousRandomPerson committed:
Requested fixes

-Used touching/turning methods from Enemy.
-Corrected managers for scene transitions.

AnonymousRandomPerson committed:
Fixed jitter when walking against a wall
